src/node_buffer.cc

```diff
@@ -1051,6 +1054,118 @@ static void EncodeUtf8String(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+static void EncodeInto(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+  Local<Context> context = env->context();
+  CHECK_GE(args.Length(), 2);
+  CHECK(args[0]->IsString());
+  CHECK(args[1]->IsUint8Array());
+
+  size_t read = 0;
+  size_t written = 0;
+
+  Utf8Value src(isolate, args[0]);
+  const char* p = *src;
+
+  Local<Uint8Array> dest = args[1].As<Uint8Array>();
+  Local<ArrayBuffer> buf = dest->Buffer();
+  char* write_result =
+      static_cast<char*>(buf->GetContents().Data()) + dest->ByteOffset();
+  size_t dest_length = dest->ByteLength();
+
+  for (size_t i = 0; i < src.length(); ) {
+    char c = p[i];
+    uint32_t code = 0;
+
+    if ((c & 0x80) == 0) {
+      code = c;
+      i += 1;
+    } else if ((c & 0xE0) == 0xC0) {
+      code = (p[i] & 0x1F) << 6;
+      code |= (p[i+1] & 0x3F);
```

Comment:
These reads can be out-of-bounds if `i` is near the `src.length()`, right?

Reply:
Because the src is UTF8, if the initial char stands for the number of the leading char, for example  if (c & 0xFC) == 0xF8) stands for there are 5 chars stand for one `Character`.If src is a valid UTF8 String,  I guess i will not be out-of-bounds.

Reply:
You’re right. If you want, feel free to add a `DCHECK()` (i.e. an assertion that only compiles in debug mode) here to make sure it really isn’t out of bounds